### PR TITLE
rc_mlx5: Add prefetching heuristic to improve non-inline recv lat

### PR DESCRIPTION
**Pull request by artpol84**
This fix provides:
* up to 2.5% improvement for medium message sizes for OSU latency.
* up to 11% (RC) and 8%(DC) improvement for OSU bi-directional bandwidth.

### CODE REVIEW

**Comment by artpol84** [comment]
@yosefe @alex-mikheev

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5459/ for details (Mellanox internal link).

**Comment by artpol84** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5469/ for details (Mellanox internal link).

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 1]
```diff
@@ -46,6 +46,7 @@ typedef struct uct_rc_mlx5_iface_common {
     struct {
         uct_ib_mlx5_cq_t   cq;
         uct_ib_mlx5_srq_t  srq;
+        void *pref_ptr;
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: indentation

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 2]
```diff
@@ -98,6 +99,23 @@ uct_rc_mlx5_iface_common_rx_inline(uct_rc_mlx5_iface_common_t *mlx5_iface,
...
+    ucs_prefetch(iface->rx.pref_ptr);
+}
+
+static UCS_F_ALWAYS_INLINE void
+uct_rc_mlx5_srq_prefetch_setup(uct_rc_mlx5_iface_common_t *iface,
+                               uct_rc_iface_t *rc_iface)
+{
+    unsigned wqe_ctr = (iface->rx.srq.free_idx + 2) & iface->rx.srq.mask;
+    uct_ib_mlx5_srq_seg_t *seg;
+    seg = uct_ib_mlx5_srq_get_wqe(&iface->rx.srq, wqe_ctr);
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: need space line before

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 3]
```diff
@@ -98,6 +99,23 @@ uct_rc_mlx5_iface_common_rx_inline(uct_rc_mlx5_iface_common_t *mlx5_iface,
...
+    uct_ib_iface_recv_desc_t *desc = seg->srq.desc;
+    iface->rx.pref_ptr = (void*)uct_ib_iface_recv_desc_hdr(&rc_iface->super, desc);
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: no need for explicit cast to void*

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 4]
```diff
@@ -98,6 +99,23 @@ uct_rc_mlx5_iface_common_rx_inline(uct_rc_mlx5_iface_common_t *mlx5_iface,
                                 byte_len);
 }
 
+static UCS_F_ALWAYS_INLINE void
+uct_rc_mlx5_srq_prefetch_first(uct_rc_mlx5_iface_common_t *iface)
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: s/iface/mlx5_common_iface/

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 5]
```diff
@@ -98,6 +99,23 @@ uct_rc_mlx5_iface_common_rx_inline(uct_rc_mlx5_iface_common_t *mlx5_iface,
...
+{
+    ucs_prefetch(iface->rx.pref_ptr);
+}
+
+static UCS_F_ALWAYS_INLINE void
+uct_rc_mlx5_srq_prefetch_setup(uct_rc_mlx5_iface_common_t *iface,
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: s/iface/mlx5_common_iface/

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 6]
```diff
@@ -151,6 +172,10 @@ uct_rc_mlx5_iface_common_poll_rx(uct_rc_mlx5_iface_common_t *mlx5_common_iface,
         hdr = uct_ib_iface_recv_desc_hdr(&rc_iface->super, desc);
         VALGRIND_MAKE_MEM_DEFINED(hdr, byte_len);
         flags = UCT_CB_PARAM_FLAG_DESC;
+        /* Assuming that next packet likely will be non-inline
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: pls add "," or "-" in the end of this line

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5491/ for details (Mellanox internal link).

**Comment by artpol84** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5492/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5493/ for details (Mellanox internal link).

**Comment by artpol84** [comment]
@yosefe
Addressed, please have a look.

**Pull request merged**